packages/clerk-js/src/ui/UserProfile/MfaBackupCodeCreatePage.tsx

```diff
@@ -0,0 +1,181 @@
+import { useClipboard } from '../hooks';
+import { mqu } from '../styledSystem';
+import { getIdentifier, handleError } from '../utils';
+import { FormButtonContainer } from './FormButtons';
+import { NavigateToFlowStartButton } from './NavigateToFlowStartButton';
+import { ContentPage } from './Page';
+
+export const MfaBackupCodeCreatePage = () => {
+  const user = useCoreUser();
+  const card = useCardState();
+  const [backupCode, setBackupCode] = React.useState<BackupCodeResource | undefined>(undefined);
+
+  const title = 'Add multi-factor authentication';
+  const text =
+    'Backup codes are now enabled. When signing in, you will need to enter a backup code as an additional step. Each code can only be used once.';
+
+  React.useEffect(() => {
+    if (backupCode) {
+      return;
+    }
+
+    void user
+      .createBackupCode()
+      .then((backupCode: BackupCodeResource) => setBackupCode(backupCode))
+      .catch(err => handleError(err, [], card.setError));
+  }, []);
+
+  if (card.error) {
+    return <ContentPage.Root headerTitle={title} />;
+  }
+
+  if (!backupCode) {
+    return (
+      <Spinner
+        colorScheme='primary'
+        size='lg'
+      />
+    );
+  }
+
+  return (
+    <ContentPage.Root headerTitle={title}>
+      <Text
+        localizationKey={text}
+        variant='regularRegular'
+      />
+      <MfaBackupCode backupCodes={backupCode.codes} />
+
+      <FormButtonContainer>
+        <NavigateToFlowStartButton
+          variant='solid'
+          autoFocus
+          localizationKey={localizationKeys('userProfile.formButtonPrimary__finish')}
+        />
+      </FormButtonContainer>
+    </ContentPage.Root>
+  );
+};
+
+type MfaBackupCodeProps = {
+  backupCodes?: string[];
+};
+
+const MfaBackupCode = (props: MfaBackupCodeProps) => {
+  const { backupCodes } = props;
+  const { applicationName } = useEnvironment().displayConfig;
+  const user = useCoreUser();
+  const { print, printableProps } = usePrintable();
+  const { onCopy, hasCopied } = useClipboard(backupCodes?.join(',') as string);
+
+  const onDownloadTxtFile = () => {
+    const element = document.createElement('a');
+    const file = new Blob([txtFileContent(backupCodes, applicationName, getIdentifier(user))], {
+      type: 'text/plain',
+    });
+    element.href = URL.createObjectURL(file);
+    element.download = `${applicationName}_backup_codes.txt`;
+    document.body.appendChild(element);
+    element.click();
+  };
+
+  if (!backupCodes) {
+    return null;
+  }
+
+  return (
+    <>
+      <Col gap={1}>
+        <Text
+          localizationKey={'Backup codes'}
+          variant='regularMedium'
+        />
+        <Text
+          localizationKey={'Store them securely and keep them secret.'}
+          variant='smallRegular'
+          colorScheme='neutral'
+        />
+      </Col>
+      <Grid
+        gap={2}
+        sx={t => ({
+          gridTemplateColumns: `repeat(5, minmax(${t.sizes.$12}, 1fr))`,
```

Comment:
❓ We'll test more on staging, but does this resolve the responsiveness issue we noticed locally?

Reply:
Yeap, did some testing and seems to work find in every screen. We will test it more on staging as you said and see if there is any issue to resolve